### PR TITLE
feat: discv5 filter peer by capability 

### LINKED ISSUE #2179: feat(discv5): filter peer by capability
### Problem
Discv5 filtering doesn't follow the RFC and filter ENR caps.

### Suggested solution
Add to the filtering a check for empty caps.

### Additional context
Follow the RFC https://rfc.vac.dev/spec/31/#usage-1

### Acceptance criteria
- [ ] Peers discovered with ENR missing the `waku2` field or no capability set must be filtered out.

### PR DESCRIPTION
**Pull request by SionoiS**
# Description
RFC#31 specify that discv5 must filter out ENRs that don't have `waku2` field or empty capability list.

# Changes
- [x] Filter predicate now reject peers with empty capability list.
- [x] Fixed test
- [x] Refactor `shardingPredicate` proc

Closes #2179

### CODE REVIEW

**Review by gabrielmer — APPROVED** [review]
Looks great 🔥 Thank you!

File: tests/test_waku_discv5.nim [hunk 1]
```diff
@@ -320,6 +320,7 @@ procSuite "Waku Discovery v5":
 
       var builder = EnrBuilder.init(enrPrivKey, seqNum = enrSeqNum)
       require builder.withWakuRelaySharding(shardsTopics).isOk()
+      builder.withWakuCapabilities(Relay)
```
Comment (Ivansete-status) [inline tests/test_waku_discv5.nim:323]: Just for me to understand better. Is this directly related to the change or why is not needed now?

**Review by Ivansete-status — APPROVED** [review]
Great catch! Thanks for it!

File: tests/test_waku_discv5.nim (hunk 1 above)
Comment (SionoiS) [reply tests/test_waku_discv5.nim:323]: When creating ENRs for tests, without specifying a Capability discv5 will reject it as invalid.

Comment (Ivansete-status) [reply tests/test_waku_discv5.nim:323]: ah yes! lovely! makes sense :)
